src/Ng/Jobs/CatalogUtility.cs

```diff
@@ -79,9 +79,18 @@ public static HttpClient CreateHttpClient(bool verbose)
             var result = new SortedList<DateTime, IList<PackageDetails>>();
 
             XElement feed;
-            using (var stream = await client.GetStreamAsync(uri))
+            try
             {
-                feed = XElement.Load(stream);
+                using (var stream = await client.GetStreamAsync(uri))
+                {
+                    feed = XElement.Load(stream);
+                }
+            }
+            catch (TaskCanceledException tce)
+            {
+                // If the client gets disposed because the HTTP request timed out, a TaskCanceledException will be thrown.
+                // It is not necessary to restart the process in this case.
+                throw new TransientException("Http client was disposed.", tce);
```

Comment:
I like the subclass idea. In this case it was httpclient request timeout. Implemented a new exception type and added more information to make easier the debug scenario.